### PR TITLE
Ability to change Encoder & Decoder for JSON & JSONB

### PR DESCRIPTION
**Pull request by c1wren**

### CODE REVIEW

File: Sources/PostgresNIO/Utilities/PostgresJSONCoder.swift [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Foundation
+
+public struct PostgresJSONCoder {
+    public static var global: PostgresJSONCoder = .init()
```
Comment (0xTim) [inline Sources/PostgresNIO/Utilities/PostgresJSONCoder.swift:4]: Are these thread safe? I'm pretty sure they're not on Linux

Comment (c1wren) [reply Sources/PostgresNIO/Utilities/PostgresJSONCoder.swift:4]: This code is almost a copy/paste from https://github.com/vapor/vapor/blob/master/Sources/Vapor/Content/ContentConfiguration.swift 

So if it isn't thread safe, then vapor isn't thread safe either.

Comment (tanner0101) [reply Sources/PostgresNIO/Utilities/PostgresJSONCoder.swift:4]: `JSONEncoder` and `JSONDecoder` are just wrapper types that initialize `_JSONEncoder` and `_JSONDecoder` respectively on each call.

**Comment by c1wren** [comment]
> Hi @c1wren, thanks for the PR. Would you mind clarifying what the motivation behind this PR is? For example, what the use case is or what problem it is trying to solve. That will help me to review it properly.

I forgot to cancel this PR. The changes in Fluent/database stuff in vapor 4 don't allow for the performance and capabilities that I had in vapor 3, so I'm migrating away from vapor altogether.

**Pull request closed without merging**